phpseclib/Net/SSH2.php

```diff
@@ -4152,4 +4164,42 @@ function setWindowSize($columns = 80, $rows = 24)
         $this->windowColumns = $columns;
         $this->windowRows = $rows;
     }
+
+    /**
+     * @return string
+     */
+    public function __toString()
+    {
+        return $this->getResourceId();
+    }
+
+    /**
+     * @return string
+     */
+    public function getResourceId()
+    {
+        return '{' . spl_object_hash($this) . '}';
```

Comment:
`{}` symbols that should not be in URL according to [RFC](http://tools.ietf.org/html/rfc3986#section-2). Same symbols Symfony2 use for place holders in URL routes. So it should safe us from any conflicts, because otherwise regexp will match all alphanumeric domains.